framework/db/BaseActiveRecord.php

```diff
@@ -1571,4 +1571,20 @@ public function offsetUnset($offset)
             unset($this->$offset);
         }
     }
+
+    /**
+     * @inheritdoc
+     *
+     * This method sets the optimistic lock value to null before calling
+     * the parent implementation.
+     */
+    public function load($data, $formName = null)
+    {
+        $lock = $this->optimisticLock();
+        if ($lock !== null) {
```

Comment:
@samdark 
The documentation of BaseActiveRecord::optimisticLock() says:
> Add a required validation rule for the version column to ensure the version value is submitted.

So maybe this line should be
```php
if ($lock !== null && $this->isAttributeRequired($lock)`) {
```
And looks like I should have added the `['version', 'required']` rule to the Document class. Am I right?